python-checks/src/main/java/org/sonar/python/checks/tests/SkippedTestNoReasonCheck.java

```diff
@@ -38,19 +41,23 @@ public class SkippedTestNoReasonCheck extends PythonSubscriptionCheck {
 
   private static final String MESSAGE = "Provide a reason for skipping this test.";
 
+  private static final Set<String> skipDecoratorsFQN = new HashSet<>(List.of("unittest.case.skip", "pytest.mark.skip"));
+  private static final Set<String> skipCallExpressionsFQN = new HashSet<>(List.of("pytest.skip"));
+
   @Override
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.DECORATOR, ctx -> {
       Decorator decorator = (Decorator) ctx.syntaxNode();
-
-      checkDecoratorSkipWithoutReason(ctx, decorator, "unittest.case.skip");
-      checkDecoratorSkipWithoutReason(ctx, decorator, "pytest.mark.skip");
+      for (String skipDecoratorFQN : skipDecoratorsFQN) {
```

Comment:
This is not what I had in mind: I think the `checkDecoratorSkipWithoutReason` should be called only once.

When you're comparing the actual decorator name with the one you provide as argument (on line 72), just try to see if you can find the name in the set instead. That way you wouldn't have to perform the same operation for each case (granted, there are only 2 and the code doesn't have to do much, so there is not really a problem performance-wise, but I think it's still a bit cleaner).

Reply:
I see ! That's indeed better, that's also what I implemented for SONARPY-766.
This has far more sense, will implement this.